src/Expectation.php

```diff
@@ -118,6 +118,38 @@ public function dd(mixed ...$arguments): void
         exit(1);
     }
 
+    /**
+     * Dump the expectation value when the result of the condition is truthy.
+     *
+     * @param  bool  $boolean
+     * @return never
+     */
+    public function ddWhen($boolean, mixed ...$arguments): void
+    {
+        if (! $boolean) {
+            return;
+        }
+
+        if (function_exists('dd')) {
+            dd($this->value, ...$arguments);
```

Comment:
Shouldn't this call our own `->dd` internally?

Reply:
Good captcha. I've updated it. Thank you!